### DISCUSSION TITLE
Problems with Multi-cluster communication with StatefulSets Documentation

### DISCUSSION THREAD

**Discussion by Flou21** [linkerd/linkerd2#7196] [General] [answered]
### What is the issue?

Hello, I have a problem with linkerd multi cluster communication with statefulsets. I found this guide in the official documentation: https://linkerd.io/2.11/tasks/multicluster-using-statefulsets/ 
I think the documentation is simply outdated, but I don't know for sure.

I thought this would help me with my existing problem. I already created a discussion in this repository (https://github.com/linkerd/linkerd2/discussions/7144), because it's not a bug, I only don't know how to configure linkerd.

### How can it be reproduced?

Follow the guide with the new k3d version.

### Logs, error output, etc

https://gist.github.com/Flou21/0d49be4207f4b09c0a8f4e35f6785a7d

(If the output is long, please create a [gist](https://gist.github.com/) and
paste the link here.)

### Environment

- Kubernetes Version: k3s version v1.21.5-k3s2 (default)
- Cluster Environment: (GKE, AKS, kops, ...) k3d
- Host OS: Mac OS
- Linkerd version: Client version: stable-2.11.0

### Possible solution
Don't have any

### Additional context
I am looking for about two weeks for a solution for a specific problem. I already created a discussion in this repositoy
https://github.com/linkerd/linkerd2/discussions/7144

**Comment by kleimkuhler** [comment]
@Flou21 I moved this to a discussion since it doesn't seem to be about an issue with Linkerd, but more about how to setup your cluster properly.

I'm looking at the linked gist output and see `Error: unknown flag: --k3s-server-arg`. What version of `k3d` do you have installed?

For me this command does work and my version is
```
❯ k3d version
k3d version v4.4.6
k3s version v1.21.1-k3s1 (default)
```

It's possible you are on a version that doesn't have this flag as an option?

**Comment by Flou21** [comment]
this is my version of k3d

```
❯ k3d --version
k3d version v5.0.1
k3s version v1.21.5-k3s2 (default
```

**Comment by kleimkuhler** [comment]
Looking at the v5 [release notes](https://github.com/rancher/k3d/releases/tag/v5.0.0) it looks like `--k3s-server-arg` changed to `--k3s-arg`.

**Comment by Flou21** [comment]
Ok

When I change this in the `create.sh` script I get another error message.

````
❯ ./create.sh
+ export ORG_DOMAIN=cluster.local
+ ORG_DOMAIN=cluster.local
+ port=6440
+ for cluster in east west
+ k3d cluster get east
+ k3d cluster create east --api-port=6440 --network=multicluster-example --k3s-arg=--cluster-domain=east.cluster.local --wait
FATA[0000] K3sExtraArg '--cluster-domain=east.cluster.local' lacks a node filter, but there's more than one node
````

I will try resolving this issue later today, when I come back from work.
I wanted to tell that a guide in the official documentation seems outdated.

**Reply by kleimkuhler** [reply]
Yep thanks for pointing out that it is now outdated! For the node filter issue it looks like you'll need to add `server:0`: ` --k3s-arg=--cluster-domain=east.cluster.local@server:0`.

**Comment by Flou21** [comment, marked as answer]
I close this issue/discussion.
